[lldb][RPC] Upstream lldb-rpc-gen tool

chelcassanova opened the pull request:
This commit upstreams the `lldb-rpc-gen` tool, a ClangTool that generates the LLDB RPC client and server interfaces. This tool, as well as LLDB RPC itself is built by default. If it needs to be disabled, put -DLLDB_BUILD_LLDBRPC=OFF in your CMake invocation.

https://discourse.llvm.org/t/rfc-upstreaming-lldb-rpc/85804

chelcassanova committed:
[lldb][RPC] Upstream lldb-rpc-gen tool

This commit upstreams the `lldb-rpc-gen` tool, a ClangTool that
generates the LLDB RPC client and server interfaces.

https://discourse.llvm.org/t/rfc-upstreaming-lldb-rpc/85804

bulbazord reviewed (commented):
Given that I wrote some of this code, it looks mostly good to me 😄 Note that this has a dependency on the emitters so this probably shouldn't land quite yet.
I commented on some of the FIXMEs because I think now is the time to really address them. It would also be great to get a fresh pair of eyes on this...

chelcassanova commented on lldb/tools/lldb-rpc/lldb-rpc-gen/RPCCommon.cpp:
```diff
@@ -0,0 +1,535 @@
+    // (thread_t) that is not `void *`, so special casing it is more effort than
+    // it's worth.
+    "_ZN4lldb8SBHostOS10ThreadJoinEP17_opaque_pthread_tPPvPNS_7SBErrorE",
+    "_ZN4lldb8SBHostOS12ThreadCancelEP17_opaque_pthread_tPNS_7SBErrorE",
```
We should be able to, it would change `MethodIsDisallowed` to just check for some underlying `DeprecatedAttr` instead of reading from this list.

chelcassanova replied:
Trying that idea out, it seems that just checking the attributes on a given method to see if it has a deprecated attribute isn't working? I'm wondering if it's because these methods are marked as deprecated using a preprocessor macro that wraps around the `[[deprecated]]` attribute itself... 🤔

chelcassanova replied:
Adding the `[[deprecated]]` attribute does allow the approach of just checking the attributes with Clang to work as intended so we'll probably have to do some preprocessor reading to make sure that we're capturing that `LLDB_DEPRECATED` macro.

chelcassanova replied:
@bulbazord Actually, I think this preprocessor macro has its link to the `[[deprecated]]` attribute removed in SBDefines if we're generating the SWIG bindings:
```
#if defined(SWIG) || _cplusplus < 201402L
#undef LLDB_DEPRECATED
#undef LLDB_DEPRECATED_FIXME
#define LLDB_DEPRECATED(MSG)
#define LLDB_DEPRECATED_FIXME(MSG, FIX)
#endif
```

and we're always generating the SWIG bindings when building RPC because of testing right? The definition for `LLDB_DEPRECATED` that ties it to the compiler's deprecated attribute is coming from `lldb-defines.h` and is overridden in `SBDefines.h` to just be blank (IIUC?). I think this means that when `lldb-rpc-gen` is reading any function that has `LLDB_DEPRECATED`, there's no actual `[[deprecated]]` attribute for the tool to pick up on.

I don't currently know why we don't add the deprecated attribute when generating SWIG bindings. If that's something that cannot change, is there a better way to work around this? If not then keeping the list and adding a note; as well as adding a check for the deprecated attribute anyways in `MethodIsDisallowed` would be the way to go here.

bulbazord replied:
lldb-rpc-gen should not satisfy `defined(SWIG)` so you should have access to the `[[deprecated]]` annotation.

chelcassanova commented on lldb/tools/lldb-rpc/lldb-rpc-gen/RPCCommon.cpp:
```diff
@@ -0,0 +1,535 @@
+    "SBProgress",
+};
+
+static constexpr llvm::StringRef MethodsWithPointerPlusLen[] = {
```
Ismail had an idea here to use a typedef over methods that used pointers and lens so that the tool could pick it up and I liked that idea. Otherwise, if we have a function that has a pointer and a length then we have no way of knowing whether or not they're related to each other.

chelcassanova replied:
I misspoke here, it wouldn't be a typedef but a new SB class that holds a struct that would be used in place of pointer + lens. This class could then be picked up by the tool (i.e. if one of the function params uses this class) to know if it's a pointer + len method without using this list.

bulbazord replied:
I'm still not sure what this would look like. Do you have some pseudocode or a sketch of the idea? I'm hesitant to add an entire new class just for RPC's sake, but it's hard to evaluate without more details.

chelcassanova replied:
So you'd have a class called something like `SBBuffer` and it would have a struct:
```
struct Buffer {
    Buffer(void *buf, size_t size) {}
};
```

that covers a pointer + len. In a function that uses it (e.g. `size_t ReadRawData(lldb::SBError &error, lldb::offset_t offset, void *buf, size_t size`) becomes `size_t ReadRawData(lldb::SBError &error, lldb::offset_t offset, SBBuffer buf)`. 

The tool could see that the method has an `SBBuffer` parameter, so that could satisfy the "this method is using a pointer followed by a length". The older version of the original method would be marked as deprecated.

Obviously the main task here is having a cleaner way of knowing this detail about methods without maintaining an ever growing list. Another path we could try is using Clang's `annotate` attribute to note this information on the API side. The tool should also be able to pick up on this, but it also means adding an attribute to the main APIs that isn't very meaningful if you're not using RPC.

bulbazord replied:
I see. I think something like that could work, but it'd be a larger change that would require buy-in from the larger community and an audit of the SBAPI in general.

chelcassanova replied:
@bulbazord Restarting this discussion because it's important that we try and figure out a way to not rely on this list in the generator anymore. This is still one of the parts of the generator that's maintained by hand and when a method isn't added to this list it leads to problems within the generated files.

I'm honestly not sure of how to try and solve this problem without touching the main SBAPI files. I had talked about using a new class to represent a pointer + len above, another idea I had had is to use an "annotation" attribute from Clang: https://clang.llvm.org/docs/AttributeReference.html#annotate which would also require the main SB API headers to be touched. This attribute takes a string and we could have something like

```
[[clang::annotate("LLDBRPC: Pointer plus len")]]
int SBFoo();
```

since the generator should be able to pick up this attribute and act on it. Adding a compiler attributes to the header files for RPC specifically might not be the best thing to do since those who don't know about RPC wouldn't know what the attribute is for however.

I'd love to get your insight on this.

DavidSpickett replied:
Assuming I forgot what RPC was, the way I'm going to be inconvenienced by this is a build error with some new SB API I add. Right?

If that error is consistent enough, we could add a section in the RPC documentation like:
```
# Pointer Plus Length Annotations

When you see an error like:
<the general form of the error, in a way that a decent "find in files" would find this>

You may need to add an annotation to tell the RPC generator that bla bla bla
```
Then, if you search for the base text of the error message, or for LLDBRPC: Pointer Plus Len, you'll find the documentation and know what to do, or why the annotation was added to something you're researching.

Whether an annotation is a good idea technically, idk, but I think the above solves the discovery problem as best we can.

chelcassanova commented on lldb/tools/lldb-rpc/lldb-rpc-gen/RPCCommon.cpp:
```diff
@@ -0,0 +1,535 @@
+// FIXME: Change the implementation of these methods so that they can be given a
+// connection parameter.
+static constexpr llvm::StringRef
+    MethodsThatUnconditionallyDoNotNeedConnection[] = {
```
Meaning that we'd only keep this downstream?

chelcassanova commented on lldb/tools/lldb-rpc/lldb-rpc-gen/RPCCommon.cpp:
```diff
@@ -0,0 +1,535 @@
+// lldb-rpc-gen emits classes as LocalObjectRefs by default.
+//
+// FIXME: Does it matter which one it emits by default?
+static constexpr llvm::StringRef ClassesThatInheritFromObjectRef[] = {
```
From how I understand that they work, all `LocalObjectRef`s could possibly wrapped up into `ObjectRef`s to avoid having 2 classes here.

chelcassanova commented on lldb/tools/lldb-rpc/lldb-rpc-gen/RPCCommon.cpp:
```diff
@@ -0,0 +1,546 @@
+// why they exist.
+static constexpr llvm::StringRef DisallowedClasses[] = {
+    "SBCommunication",          // What is this used for?
+    "SBInputReader",            // What is this used for?
```
@DavidSpickett On the original and larger patch that I had up for this, you asked if we could answer what these files are used for. I think Greg Clayton might know? But otherwise, these comments should be changed to better clarify that we just don't need these classes in RPC for now.

DavidSpickett replied:
@clayborg do you recall?

I don't mind if these comments aren't in depth but at least make them honest:
```
// No one ever asked for this and to be honest we don't even know why it exists in the SB API in the first place.
```
Is fine, as long as it's the truth :)

chelcassanova commented on lldb/tools/lldb-rpc/lldb-rpc-gen/RPCCommon.cpp:
```diff
@@ -0,0 +1,535 @@
+};
+
+static llvm::StringMap<llvm::SmallVector<llvm::StringRef>>
+    ClassName_to_ParameterTypes = {
```
We actually shouldn't need this map anymore. It was originally created to create a new SB object in the event that an incoming one had an invalid connection. We assert on SB objects with bad connections now. I'm gonna remove this and the function that's using it from this patch.

chelcassanova commented on lldb/tools/lldb-rpc/lldb-rpc-gen/RPCCommon.cpp:
```diff
@@ -0,0 +1,535 @@
+  return HasCallbackParameter && HasBatonParameter;
+}
+
+// FIXME: Find a better way to do this. Here is why it is written this way:
```
So I have a couple thoughts on this FIXME. Just to clarify the main thought from your original comment, since we're picking up all our info from `liblldb`, qualified types for methods and their return types/params have the `lldb::` namespace and all references to this need to change to using `lldb_rpc::`. Currently, we change these namespaces as needed using this replace namespace function that we've implemented, but we'd rather replace those namespaces upon creating a `Method` object instead of when we reference information from `Method`.

I like the idea of using Clang's `ChangeNamespace` tool instead of doing our own string replacement here, but looking at some example code for it it looks like it operates at the file level. (i.e. it takes a pattern of files to do the namespace changing). If we were to try using this, that gives me the initial idea of allowing `lldb-rpc-gen` to write everything with the original `lldb::` namespace and then using `ChangeNamespace` once everything has been said and done to replace the namespaces _after_ `lldb-rpc-gen` has emitted everything.

I think we could also keep this `ReplaceLLDBNamespaceWithRPCNamespace` method, but possibly just use it to replace the strings that `Method` uses to hold on to the qualified name. This should reduce some of the places it's being used, but doesn't address return types or params since we hold on to those as `QualTypes` which is going to have the original namespaces.

It would be nice if the namespaces were already changed when coming into `lldb-rpc-gen`, which I think that `ChangeNamespace` could do, but it would apply that change to the original SB API files? That would possibly mess up the info we need in other unsavoury ways.

bulbazord replied:
I think operating in 2 passes could be a good idea here. It was a bit of whack-a-mole getting the `ReplaceLLDBNamespaceWithRPCNamespace` stuff all working as intended, so it'd greatly simplify the code and probably be easier to reason about.

The key would be to only apply it to the library. The server links against LLDB directly.

chelcassanova replied:
> I think operating in 2 passes could be a good idea here.

If we're ok with having a dependency on `clang-tools-extra`, then we could use the `clang-change-namespace` tool itself after the files have been generated. Otherwise, if we want to reimplement the code from `ChangeNamespace` in `lldb-rpc-gen`, that _looks_ like it should be feasible? Again, I'm just looking at the code from the change namespace stuff but I think it's backed by `Refactoring` in Clang tooling. We already have that dependency, I think the change namespace code is a specific application of `RefactoringTool`.

This could be worth pursuing. It's not just that this cuts down on our string replacement function, but this might also be useful for cutting out the [Python scripts](https://github.com/llvm/llvm-project/pull/138028) that we're using to modify the namespaces for the lldb headers.

> The key would be to only apply it to the library.

Correct. As it is right now, `lldb-rpc-gen` generates all server, client and byproduct code within the same file, I wonder if this lends credence to subtooling this? :)

DavidSpickett commented on lldb/tools/lldb-rpc/lldb-rpc-gen/RPCCommon.cpp:
```diff
@@ -0,0 +1,502 @@
+                                // really needed though.
+    "SBCommand", // There's nothing too difficult about this one, but many of
+                 // its methods take a SBCommandPluginInterface pointer so
+                 // there's no reason to support this.
```
Needs extending to explain why `SBCommandPluginInterface` means there's no reason to support it. Like:
```
SBCommandPluginInterface is no use in an lldb-rpc context because X Y and Z
```

DavidSpickett commented on lldb/tools/lldb-rpc/lldb-rpc-gen/RPCCommon.cpp:
```diff
@@ -0,0 +1,502 @@
+// we use for testing. Because of this, there is no annotation for the tool to
+// pick up on so this list will be used while we have this restriction in
+// SBDefines.h.
+static constexpr llvm::StringRef DisallowedMethods[] = {
```
Add a comment here to explain why mangled names are used. You explained this to me somewhere else and I know I agree with the choice, but I forget the details.

DavidSpickett commented on lldb/tools/lldb-rpc/lldb-rpc-gen/RPCCommon.cpp:
```diff
@@ -0,0 +1,502 @@
+
+QualType lldb_rpc_gen::GetUnqualifiedUnderlyingType(QualType T) {
+  QualType UnderlyingType = GetUnderlyingType(T);
+  return UnderlyingType.getUnqualifiedType();
```
Combine these two into one line, unless you intended to error check the first call in which case do that.

chelcassanova replied:
> unless you intended to error check the first call in which case do that.

I don't think we did, so I can change this.

DavidSpickett commented on lldb/tools/lldb-rpc/lldb-rpc-gen/RPCCommon.cpp:
```diff
@@ -0,0 +1,502 @@
+
+bool lldb_rpc_gen::TypeIsFromLLDBPrivate(QualType T) {
+
+  auto CheckTypeForLLDBPrivate = [](const Type *Ty) {
+    if (!Ty)
+      return false;
+    const auto *CXXRDecl = Ty->getAsCXXRecordDecl();
+    if (!CXXRDecl)
+      return false;
+    const auto *NSDecl =
+        llvm::dyn_cast<NamespaceDecl>(CXXRDecl->getDeclContext());
+    if (!NSDecl)
+      return false;
+    return NSDecl->getName() == "lldb_private";
```
This should be a `static` function defined just before this function.

And when you do that, delete the empty line at the start of TypeIsFromLLDBPrivate.

DavidSpickett commented on lldb/tools/lldb-rpc/lldb-rpc-gen/RPCCommon.cpp:
```diff
@@ -0,0 +1,502 @@
+  bool HasBatonParameter = false;
+  auto End = MDecl->parameters().end();
+  for (auto Iter = MDecl->parameters().begin(); Iter != End; Iter++) {
+    if ((*Iter)->getType()->isFunctionPointerType()) {
+      HasCallbackParameter = true;
+      continue;
+    }
+
+    if ((*Iter)->getType()->isVoidPointerType())
+      HasBatonParameter = true;
```
This would make more sense if it was:
if (is function pointer type)
  // is a callback parameter
else if (is void pointer type)
  // is a baton

DavidSpickett commented on lldb/tools/lldb-rpc/lldb-rpc-gen/RPCCommon.cpp:
```diff
@@ -0,0 +1,502 @@
+  auto Pos = Name.find("lldb::");
+  while (Pos != std::string::npos) {
+    constexpr size_t SizeOfLLDBNamespace = 4;
+    Name.replace(Pos, SizeOfLLDBNamespace, "lldb_rpc");
```
This is confusing because "lldbnamespace" can be interpreted as "lldb" or "lldb::". And you search for one of them and then replace the other.

Would be clearer as:
```
  auto Pos = Name.find("lldb::");
  while (Pos != std::string::npos) {
    Name.replace(Pos, /*count=*/4 /* "lldb" */, "lldb_rpc");
```

Same deal for StripLLDBNamespace.

chelcassanova replied:
> This is confusing because "lldbnamespace" can be interpreted as "lldb" or "lldb::".

True, when we use this function we're using it to repalce instances of "lldb::" (same for `StripLLDBNamespace`)

> Or you could replace lldb:: with lldb_rpc:: and that way you are replacing exactly what you search for, and get the length at compile time.

I like this approach, it's clearer exactly what we're doing here.

DavidSpickett commented on lldb/tools/lldb-rpc/lldb-rpc-gen/RPCCommon.cpp:
```diff
@@ -0,0 +1,502 @@
+
+bool lldb_rpc_gen::Method::operator<(const lldb_rpc_gen::Method &rhs) const {
+  return this < &rhs;
+}
```
What's the ordering here? Appears to be memory address.

chelcassanova replied:
It should be on memory address, it's used so that we can have `Method`s in ordered containers (mainly `set`s).

DavidSpickett commented on lldb/tools/lldb-rpc/lldb-rpc-gen/RPCCommon.h:
```diff
@@ -0,0 +1,107 @@
+
+class FileEmitter {
+protected:
+  FileEmitter(std::unique_ptr<llvm::ToolOutputFile> &&OutputFile)
```
Unused in this PR, but used in later ones?

chelcassanova replied:
The `FileEmitter` class is used in the emitter classes, like the server emitter (https://github.com/llvm/llvm-project/pull/138032) so I think should stay.

DavidSpickett commented on lldb/tools/lldb-rpc/lldb-rpc-gen/lldb-rpc-gen.cpp:
Could be folded into the line below, and add the parameter name like `/*filename=*/"LLDBRPC.h"`.

DavidSpickett commented on lldb/tools/lldb-rpc/lldb-rpc-gen/lldb-rpc-gen.cpp:
You can do:
```
AmalgamatedLibraryHeader->os() << "#ifndef LLDBRPC_H\n";
 << "other stuff"
  << "more stuff";
```
To reduce the repetition.

DavidSpickett replied:
General idea applies to EmitClassNamesFile as well.

chelcassanova replied:
Just seeing this, I'll apply it to this patch. I'm actually kind of surprised that clangd doesn't already do something like this 🤔

DavidSpickett commented on lldb/tools/lldb-rpc/lldb-rpc-gen/lldb-rpc-gen.cpp:
No braces needed for a one line for.

DavidSpickett reviewed (commented):
I don't have any objections to the way this works, just some code style.

I don't know enough about the FIXMEs to make you fix them or not, if you can that's great, otherwise consider accepting that they will never be fixed (as is the fate of the average FIXME) and convert them into informative comments instead.

DavidSpickett reviewed (commented):
I resolved all the comments that are done, I have a few left open that need addressing.

Then decide with @bulbazord what you do or don't do with the finer details. Everything else is looking good.

chelcassanova commented:
@bulbazord I have a auggestion for this patch. I want to have the minimum CMake necessary to let the tool build. To accomplish this, I want to take out all code in this tool that relates to the server/client code, leaving the code that generates the byproducts. I'd then shift the server code in `lldb-rpc-gen.cpp` to the patch that's already up for the server emitter and the client code for the patch for the client emitters. In effect:

- This patch would have all RPCCommon code for lldb-rpc-gen
- `lldb-rpc-gen.cpp` will be stripped down to the code that generates the byproducts (namely the class names, method names and skipped methods file)
- As mentioned above, the code in lldb-rpc-gen.cpp that relates to generating the server code would move to the patch that's up for the server emitters.
- A test could be put up to just check that the byproducts files have been emitted in their expected locations.

I'd love to get your thoughts on this.

chelcassanova commented:
Cool, I updated the patch to do everything I said in that comment. `lldb-rpc-gen`'s executable should now be built in the build dir.

bulbazord reviewed (commented):
Some comments. Overall looks ok to me, but I'm already familiar with most of it.

chelcassanova commented on lldb/tools/lldb-rpc/CMakeLists.txt:
```diff
@@ -0,0 +1,22 @@
+include(CheckCXXCompilerFlag)
```
I'm actually not sure, though it seems to have originally been added to silence compiler warnings, so maybe this should be kept.

chelcassanova commented:
One more time, I've updated the patch. Some points:

- I've removed our exception list for the methods that were unconditionally given a connection
- For the FIXME that we had for changing the LLDB namespace I've changed to a note as to why we use simple string replacement.
- Again, for the pointer plus len list, that can be addressed in a follow up PR.
- I added a CMake variable to optionally skip building LLDB RPC.

From this, I think this patch is pretty much ready to land if no one else finds big outstanding issues :)

bulbazord reviewed (approved):
Let's get this patch in and iterate on the tool. Please watch the bots after landing.

chelcassanova commented:
When I updated the patch, the test now fails because of the error that it's missing the clang resource dir. That repros locally and I haven't changed anything about how the tool is built outside of adding a CMake variable, investigating.

chelcassanova commented:
Actually, the test fails locally for me. On the pre-commit CI it's error message is `home/gha/actions-runner/_work/llvm-project/llvm-project/build/tools/lldb/test/Shell/RPC/Generator/Tests/Output/CheckRPCGenToolByproducts.test.script: line 1: fg: no job control`, which means that it doesn't recognize that the tool's binary is there.

As an aside, is there a way to retrigger a pre-commit CI check without having to push again?

DavidSpickett commented on lldb/test/Shell/RPC/Generator/lit.local.cfg:
```diff
@@ -0,0 +1,6 @@
+# As far as I can see, you need to check all possible ways to
+# say "no" in a CMake option.
+negation_options = ["OFF", "NO", 0]
+if not config.lldb_has_lldbrpc or config.lldb_has_lldbrpc in negation_options:
```
There is a list of variables in lldb/test/CMakeLists.txt that are passed to llvm_canonicalize_cmake_booleans. I think you need to add the new cmake option to that, then you'll not need to do this.

chelcassanova commented:
I'm still seeing the `no job control` error in the pre-commit CI. As far as I know this error is not supposed to be happening, RPC should be getting built which should mean that the `lldb-rpc-gen` tool is available for use in the shell test. I can't repro locally, I'm wondering if this is a Linux-specific issue.

I'm also wondering if we should just set the bool to enable RPC to OFF by default, land this patch and then investigate after.

DavidSpickett commented:
Also it's a bit misleading, this is printed but I don't know if it is the cause of the include problem:
```
fixed-compilation-database: Error while opening fixed database: No such file or directory
json-compilation-database: Error while opening JSON database: No such file or directory
Running without flags.
```
I expect not because the compilation database only deals with sources, .cpp, .cc etc. not headers.

chelcassanova commented:
> I expect not because the compilation database only deals with sources, .cpp, .cc etc. not headers.

So it should be trying to load something like `compile_commands.json`? Which is in the root of the build dir so when the tool needs to load it's going to try and do some from the test dir which is not where that file would be.

chelcassanova commented:
For the include problem, `cstdint` isn't in the Clang resource directory that gets passed in. maybe a system include dir must also be passed in here. One thing I'm confused about is that the tool is built and run correctly in the actual CMake build phase.

chelcassanova commented:
Something I just noticed is that we're supposed to pass a build path to the tool with `-p` which is done in the actual build but not for the test. I'm updating the test to do that, which should resolve the compilation database issue. Also, instead of using an input path from source I'll use a dummy input file.

chelcassanova commented:
@DavidSpickett I think this is ready to go, I'd love if you could take another look at this.

zeroomega commented:
Hi, we noticed that after 68c8c8ceeba6da96189335236e3ec80a082e4d7b got landed, the lldb test start to have build issues on linux-x64.  It failed the when `lldb/tools/lldb-rpc/generated/lldb.py` try to generate sources and complain cstdint not found. Could you take a look please? I suspect maybe the sysroot used by CMake is not correctly propergated lldb-rpc-gen correctly.

Error message:

```
[836/846](2) Building CXX object tools/lldb/tools/lldb-rpc-gen/CMakeFiles/lldb-rpc-gen.dir/lldb-rpc-gen.cpp.o
/b/s/w/ir/x/w/llvm-llvm-project/lldb/tools/lldb-rpc-gen/lldb-rpc-gen.cpp:105:18: warning: unused variable 'HasCallbackParameter' [-Wunused-variable]
  105 |       const bool HasCallbackParameter =
      |                  ^~~~~~~~~~~~~~~~~~~~
1 warning generated.
[837/846](2) Linking CXX executable bin/lldb-rpc-gen
[838/846](2) Generating sources for lldb-rpc-server...
FAILED: tools/lldb/tools/lldb-rpc/generated/SBClasses.def tools/lldb/tools/lldb-rpc/generated/SBAPI.def tools/lldb/tools/lldb-rpc/generated/lldb.py tools/lldb/tools/lldb-rpc/generated/server/SBAPI.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBAddress.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBAddress.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBAddressRange.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBAddressRange.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBAddressRangeList.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBAddressRangeList.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBAttachInfo.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBAttachInfo.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBBlock.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBBlock.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBBreakpoint.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBBreakpoint.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBBreakpointLocation.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBBreakpointLocation.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBBreakpointName.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBBreakpointName.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBBroadcaster.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBBroadcaster.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBCommandInterpreter.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBCommandInterpreter.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBCommandInterpreterRunOptions.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBCommandInterpreterRunOptions.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBCommandReturnObject.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBCommandReturnObject.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBCompileUnit.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBCompileUnit.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBData.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBData.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBDebugger.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBDebugger.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBDeclaration.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBDeclaration.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBEnvironment.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBEnvironment.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBError.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBError.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBEvent.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBEvent.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBExecutionContext.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBExecutionContext.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBExpressionOptions.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBExpressionOptions.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBFile.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBFile.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBFileSpec.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBFileSpec.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBFileSpecList.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBFileSpecList.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBFormat.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBFormat.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBFrame.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBFrame.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBFunction.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBFunction.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBHostOS.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBHostOS.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBInstruction.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBInstruction.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBInstructionList.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBInstructionList.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBLanguageRuntime.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBLanguageRuntime.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBLaunchInfo.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBLaunchInfo.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBLineEntry.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBLineEntry.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBListener.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBListener.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBMemoryRegionInfo.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBMemoryRegionInfo.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBMemoryRegionInfoList.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBMemoryRegionInfoList.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBModule.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBModule.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBModuleSpec.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBModuleSpec.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBMutex.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBMutex.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBPlatform.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBPlatform.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBProcess.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBProcess.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBProcessInfo.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBProcessInfo.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBProcessInfoList.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBProcessInfoList.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBProgress.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBProgress.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBQueue.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBQueue.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBQueueItem.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBQueueItem.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBReproducer.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBReproducer.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBSaveCoreOptions.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBSaveCoreOptions.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBScriptObject.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBScriptObject.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBSection.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBSection.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBSourceManager.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBSourceManager.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBStatisticsOptions.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBStatisticsOptions.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBStream.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBStream.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBStringList.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBStringList.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBStructuredData.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBStructuredData.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBSymbol.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBSymbol.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBSymbolContext.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBSymbolContext.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBSymbolContextList.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBSymbolContextList.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBTarget.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBTarget.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBThread.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBThread.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBThreadCollection.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBThreadCollection.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBThreadPlan.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBThreadPlan.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBTrace.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBTrace.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBTraceCursor.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBTraceCursor.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBType.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBType.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeCategory.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeCategory.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeEnumMember.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeEnumMember.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeFilter.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeFilter.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeFormat.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeFormat.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeNameSpecifier.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeNameSpecifier.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeSummary.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeSummary.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeSynthetic.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeSynthetic.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBUnixSignals.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBUnixSignals.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBValue.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBValue.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBValueList.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBValueList.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBVariablesOptions.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBVariablesOptions.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBWatchpoint.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBWatchpoint.cpp tools/lldb/tools/lldb-rpc/generated/server/Server_SBWatchpointOptions.h tools/lldb/tools/lldb-rpc/generated/server/Server_SBWatchpointOptions.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/SBClasses.def /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/SBAPI.def /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/lldb.py /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/SBAPI.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBAddress.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBAddress.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBAddressRange.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBAddressRange.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBAddressRangeList.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBAddressRangeList.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBAttachInfo.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBAttachInfo.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBBlock.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBBlock.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBBreakpoint.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBBreakpoint.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBBreakpointLocation.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBBreakpointLocation.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBBreakpointName.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBBreakpointName.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBBroadcaster.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBBroadcaster.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBCommandInterpreter.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBCommandInterpreter.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBCommandInterpreterRunOptions.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBCommandInterpreterRunOptions.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBCommandReturnObject.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBCommandReturnObject.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBCompileUnit.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBCompileUnit.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBData.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBData.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBDebugger.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBDebugger.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBDeclaration.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBDeclaration.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBEnvironment.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBEnvironment.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBError.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBError.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBEvent.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBEvent.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBExecutionContext.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBExecutionContext.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBExpressionOptions.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBExpressionOptions.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBFile.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBFile.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBFileSpec.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBFileSpec.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBFileSpecList.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBFileSpecList.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBFormat.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBFormat.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBFrame.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBFrame.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBFunction.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBFunction.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBHostOS.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBHostOS.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBInstruction.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBInstruction.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBInstructionList.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBInstructionList.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBLanguageRuntime.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBLanguageRuntime.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBLaunchInfo.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBLaunchInfo.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBLineEntry.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBLineEntry.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBListener.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBListener.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBMemoryRegionInfo.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBMemoryRegionInfo.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBMemoryRegionInfoList.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBMemoryRegionInfoList.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBModule.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBModule.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBModuleSpec.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBModuleSpec.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBMutex.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBMutex.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBPlatform.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBPlatform.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBProcess.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBProcess.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBProcessInfo.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBProcessInfo.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBProcessInfoList.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBProcessInfoList.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBProgress.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBProgress.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBQueue.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBQueue.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBQueueItem.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBQueueItem.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBReproducer.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBReproducer.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBSaveCoreOptions.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBSaveCoreOptions.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBScriptObject.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBScriptObject.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBSection.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBSection.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBSourceManager.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBSourceManager.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBStatisticsOptions.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBStatisticsOptions.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBStream.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBStream.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBStringList.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBStringList.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBStructuredData.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBStructuredData.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBSymbol.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBSymbol.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBSymbolContext.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBSymbolContext.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBSymbolContextList.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBSymbolContextList.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBTarget.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBTarget.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBThread.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBThread.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBThreadCollection.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBThreadCollection.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBThreadPlan.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBThreadPlan.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBTrace.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBTrace.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBTraceCursor.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBTraceCursor.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBType.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBType.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeCategory.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeCategory.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeEnumMember.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeEnumMember.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeFilter.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeFilter.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeFormat.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeFormat.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeNameSpecifier.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeNameSpecifier.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeSummary.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeSummary.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeSynthetic.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBTypeSynthetic.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBUnixSignals.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBUnixSignals.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBValue.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBValue.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBValueList.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBValueList.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBVariablesOptions.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBVariablesOptions.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBWatchpoint.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBWatchpoint.cpp /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBWatchpointOptions.h /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server/Server_SBWatchpointOptions.cpp 
cd /b/s/w/ir/x/w/llvm_build && /b/s/w/ir/x/w/cipd_tool/fuchsia/third_party/cmake/72c014f0086372a2bab7ed771e2ed942d6122d94331d19c95566bb25d2f962ff/bin/cmake -E make_directory /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated && /b/s/w/ir/x/w/cipd_tool/fuchsia/third_party/cmake/72c014f0086372a2bab7ed771e2ed942d6122d94331d19c95566bb25d2f962ff/bin/cmake -E make_directory /b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated/server && /b/s/w/ir/x/w/llvm_build/bin/lldb-rpc-gen -p /b/s/w/ir/x/w/llvm_build --output-dir=/b/s/w/ir/x/w/llvm_build/tools/lldb/tools/lldb-rpc/generated --extra-arg="-USWIG" /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBAddress.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBAddressRange.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBAddressRangeList.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBAttachInfo.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBBlock.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBBreakpoint.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBBreakpointLocation.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBBreakpointName.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBBroadcaster.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBCommandInterpreter.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBCommandInterpreterRunOptions.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBCommandReturnObject.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBCompileUnit.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBData.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBDebugger.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBDeclaration.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBEnvironment.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBError.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBEvent.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBExecutionContext.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBExpressionOptions.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBFile.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBFileSpec.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBFileSpecList.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBFormat.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBFrame.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBFunction.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBHostOS.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBInstruction.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBInstructionList.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBLanguageRuntime.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBLaunchInfo.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBLineEntry.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBListener.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBMemoryRegionInfo.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBMemoryRegionInfoList.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBModule.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBModuleSpec.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBMutex.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBPlatform.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBProcess.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBProcessInfo.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBProcessInfoList.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBProgress.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBQueue.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBQueueItem.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBReproducer.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBSaveCoreOptions.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBScriptObject.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBSection.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBSourceManager.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBStatisticsOptions.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBStream.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBStringList.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBStructuredData.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBSymbol.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBSymbolContext.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBSymbolContextList.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBTarget.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBThread.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBThreadCollection.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBThreadPlan.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBTrace.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBTraceCursor.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBType.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBTypeCategory.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBTypeEnumMember.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBTypeFilter.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBTypeFormat.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBTypeNameSpecifier.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBTypeSummary.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBTypeSynthetic.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBUnixSignals.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBValue.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBValueList.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBVariablesOptions.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBWatchpoint.h /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBWatchpointOptions.h
[1/78] Processing file /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBAddress.h.
In file included from /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBAddress.h:12:
In file included from /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/API/SBDefines.h:12:
In file included from /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/lldb-defines.h:12:
In file included from /b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/lldb-types.h:12:
/b/s/w/ir/x/w/llvm-llvm-project/lldb/include/lldb/lldb-enumerations.h:12:10: fatal error: 'cstdint' file not found
   12 | #include <cstdint>
      |          ^~~~~~~~~
1 error generated.
```
